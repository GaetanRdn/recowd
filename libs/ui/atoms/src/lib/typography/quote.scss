@mixin rc-quote {
  .rc-quote {
    font-style: italic;

    &::before,
    &::after {
      content: '"';
    }
  }
}
