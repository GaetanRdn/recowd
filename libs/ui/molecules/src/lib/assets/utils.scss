@function map-get-deep($map, $level, $subLevel) {
  @return map-get($map, $level, $subLevel);
}

@mixin use-typography($typography) {
  @each $key in font-size, font-family, font-weight, line-height {
    #{$key}: map-get($typography, $key);
  }
}
