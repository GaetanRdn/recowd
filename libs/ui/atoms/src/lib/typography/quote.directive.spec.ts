import { QuoteDirective } from './quote.directive';
import { TemplateLookup } from '@recowd/test/utils';
import { Component } from '@angular/core';
import { TestBed, waitForAsync } from '@angular/core/testing';

describe('QuoteDirective', () => {
  let templateLookup: TemplateLookup<HostComponent>;

  beforeEach(waitForAsync(() => {
    TestBed.configureTestingModule({
      declarations: [HostComponent],
      imports: [QuoteDirective],
    }).compileComponents();

    templateLookup = new TemplateLookup(HostComponent);
    templateLookup.detectChanges();
  }));

  it('should create', () =>
    expect(templateLookup.firstChildElement).toMatchSnapshot());
});

@Component({
  template: ` <rc-quote>Quote</rc-quote>`,
})
class HostComponent {}
