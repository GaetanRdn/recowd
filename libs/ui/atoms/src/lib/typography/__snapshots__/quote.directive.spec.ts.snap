// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`QuoteDirective should create 1`] = `
<rc-quote
  class="rc-quote"
>
  Quote
</rc-quote>
`;
